### PR TITLE
Skip building git gem extensions if they're already built

### PR DESCRIPTION
**Pull request by indirect**

### CODE REVIEW

**Comment by andremedeiros** [comment]
I wonder how many :+1:s I can fit in here

**Comment by andremedeiros** [comment]
@csfrancis ping

**Comment by segiddins** [comment]
Looks good to me! :+1:

**Comment by andremedeiros** [comment]
We've had some spec regressions with this fix. While I'm fairly certain that the fix is correct, we need to investigate and potentially fix the following test cases:

- [ ] bundle clean doesn't remove extensions artifacts from bundled git gems after clean `rspec ./spec/commands/clean_spec.rb:652`
- [ ] bundle exec with gems bundled via :path with invalid gemspecs outputs the gemspec validation errors `rspec ./spec/commands/exec_spec.rb:324`
- [ ] bundle package with --all with a gemspec caches all dependencies except bundler and the gemspec specified gem `rspec ./spec/commands/package_spec.rb:46`
- [ ] bundle install with git sources with an extension installs the extension `rspec ./spec/install/gemfile/git_spec.rb:863`
- [ ] bundle install with git sources with an extension does not use old extension after ref changes `rspec ./spec/install/gemfile/git_spec.rb:890`
- [ ] bundle install with git sources with an extension does not prompt to gem install if extension fails `rspec ./spec/install/gemfile/git_spec.rb:927`
- [ ] bundle install with explicit source paths warns on invalid specs `rspec ./spec/install/gemfile/path_spec.rb:145`

**Comment by csfrancis** [comment]
You all are doing God's work.

**Comment by segiddins** [comment]
I think I might have this passing now....

**Comment by segiddins** [comment]
@indirect / @andremedeiros please review

File: spec/install/gemfile/git_spec.rb [hunk 1]
```diff
@@ -943,6 +943,47 @@
       expect(out).to include("An error occurred while installing foo (1.0)")
       expect(out).not_to include("gem install foo")
     end
+
+    it "does not reinstall the extension" do
```
Comment (segiddins) [inline spec/install/gemfile/git_spec.rb]: @indirect this whole PR will be good to go once this is scoped to RubyGems > 2.3, I think

**Comment by segiddins** [comment]
@indirect r?

**Comment by indirect** [comment]
@segiddins I am r+ on this PR, but you do still need to fix the validation specs on old RubyGems versions: https://travis-ci.org/bundler/bundler/jobs/111326359#L3103

**Comment by segiddins** [comment]
_Crosses fingers_

**Comment by segiddins** [comment]
OMG finally passed!
@homu r+

**Comment by csfrancis** [comment]
:tada: thanks for all your help on this guys!

**Pull request merged**